src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java

```diff
@@ -116,7 +80,7 @@ public void accept(final RestChannel channel) throws Exception {
                 final XContentBuilder builder = channel.newBuilder();
                 BytesRestResponse response;
                 try {
-                    if (vendor == null) {
+                    if (securityTokenManager == null) {
```

Comment:
Will the securityTokenManager ever be null? It looks like it will always be present, but the internal jwtVendor is nullable.

wdyt about having an `isInitialized()` on the `securityTokenManager` and using that here?

Reply:
Good catch, this is not possible to have a null tokenManager.  I've renamed and opened up `issueOnBehalfOfTokenAllowed()` to handle this scenario since Service Accounts & OBO tokens can enabled / disabled seperately.